test/tic_tac_toe_test/human_player_test.exs

```diff
@@ -11,41 +11,51 @@ defmodule TicTacToe.HumanPlayerTest do
   end
 
   defmodule ValidatorWithOneValid do
-    def error(position) do
+    def error(position, _board) do
       unless position == 1 do
         "Invalid entry"
       end
     end
   end
 
   defmodule DisplayWithInputs do
-    defstruct state: []
+    defstruct [:input, :moves]
 
     defimpl Display do
       def output(_console, _message) do
       end
 
-      def input(%DisplayWithInputs{state: state}, _message) do
-        List.first(state)
+      def input(%DisplayWithInputs{moves: moves} = display, _message) do
+        %{display | input: List.first(moves), moves: List.delete_at(moves, 0)}
```

Comment:
Hey! This is really common pattern that you can achieve with pattern matching

```suggestion
      def input(%DisplayWithInputs{moves: [input | moves]} = display, _message) do
        %{display | input: input, moves: moves}
```

Reply:
Nice!  I forgot you showed me this trick.